shesha/supervisor/saxoPlusManager.py

```diff
@@ -132,21 +138,58 @@ def next(self):
+
+        # Turbulence always disabled on 2nd instance of COMPASS                         
+        self.second_stage.atmos.enable_atmos(False) 
+        
+        if do_control:
+            # system has already been calibrated and normal run can be done
+
+            if not (self.iterations % self.frequency_ratio):
+                # Time for first stage start of new WFS exposure. 
+                self.first_stage.reset_wfs_exposure()
+                # DM shape is updated, but no centroid, neither control is computed.
+                self.first_stage.next(do_control=False, apply_control=True,
+                                      do_centroids=False, compute_tar_psf=True) 
+                
+            elif not ((self.iterations+1) % self.frequency_ratio):
+                # Finish integration of WFS image, then compute centroids,
+                # do control to get the new commands.
+                self.first_stage.next(do_control=True, apply_control=True,
+                                      do_centroids=True, compute_tar_psf=True)
+
+            else:
+                # In the middle of a WFS integration frame. Only raytracing current
+                # turbulence phase, new DMs phase (apply control), and accumulates
+                # WFS image.
+                self.first_stage.next(do_control=False, apply_control=True,
+                                      do_centroids=False, compute_tar_psf=True)
+
+        else: # probably calibration is ongoing and no control can be computed so far
+            self.first_stage.next(do_control=False, do_centroids=True,
+                                  apply_control=True, compute_tar_psf = True)
+            
+
+        # FIRST STAGE IS DONE.
         # Get residual of first stage to put it into second stage
-        # For now, involves GPU-CPU memory copies, can be improved later if speed is a limiting factor here... 
+        # For now, involves GPU-CPU memory copies, can be improved later if speed is
+        # a limiting factor here... 
         first_stage_residual = self.first_stage.target.get_tar_phase(0)
-        self.second_stage_input[self.mpup_offset:-self.mpup_offset,self.mpup_offset:-self.mpup_offset,:] = first_stage_residual[:,:,None]
+        self.second_stage_input[self.mpup_offset:-self.mpup_offset,
+                                self.mpup_offset:-self.mpup_offset,:] = first_stage_residual[:,:,None]
         self.second_stage.tel.set_input_phase(self.second_stage_input) # 1st stage residuals sent to seconds stage simulation. 
-        # Second stage computation
-        self.second_stage.next(move_atmos=False) #"Updates the second stage siulation accordingly". 
+
+        # SECOND STAGE LOOP STARTS...
+        if do_control:
+            #"Updates the second stage simulation accordingly".                          
+            self.second_stage.next(move_atmos=False) 
+        else:
+            self.second_stage.next(move_atmos=False, do_control=False)
```

Comment:
Those lines can be merged in one : 

`self.second_stage.next(move_atmos=False, do_control=do_control)`

Reply:
Addressed by commit [d484b28](https://github.com/ANR-COMPASS/shesha/pull/44/commits/d484b284b183f9b2c04c235e5f67392db45019de)